server/client.go

```diff
@@ -1965,20 +1965,25 @@ func (c *client) removeUnauthorizedSubs() {
 	for sid, sub := range c.subs {
 		subs[sid] = sub
 	}
-	c.mu.Unlock()
+
+	var removedSubs [32]*subscription
+	removed := removedSubs[:0]
 
 	for sid, sub := range subs {
 		if c.sl != nil && !c.canSubscribe(sub.subject) {
-			_ = c.sl.Remove(sub)
-			c.mu.Lock()
+			removed = append(removed, sub)
 			delete(c.subs, sid)
-			c.mu.Unlock()
-			c.sendErr(fmt.Sprintf("Permissions Violation for Subscription to %q (sid %s)",
-				sub.subject, sub.sid))
-			srv.Noticef("Removed sub %q for user %q - not authorized",
-				string(sub.subject), c.opts.Username)
 		}
 	}
+	c.mu.Unlock()
+
+	for _, sub := range removed {
+		c.sl.Remove(sub)
```

Comment:
Or could skip this and before or after the loop call `c.sl.RemoveBatch(removed)`

Reply:
Good call.. Will change..